### DISCUSSION TITLE
How to copy alpha channel from one Tiff image to another Tiff image using Magic .Net

### DISCUSSION THREAD

**Discussion by adobris** [dlemstra/Magick.NET#631] [General] [answered]
I am pretty new in ImageMagic and I need some help.
How to copy alpha channel from one Tiff image with alpha channel to another Tiff image without alpha channel using Magic .Net.

Image with alpha:
https://drive.google.com/open?id=1u0SD7gk4s0Ynh91axs_P67alZGCfO4nu

Image without alpha:
https://drive.google.com/open?id=19XCd1blxHvJ3FVPqyx4bUfLoZ6qe3HHp

Or how to add alpha channel from PNG alpha file to a Tiff image without alpha channel using Magic .Net.

**Comment by dlemstra** [comment, marked as answer]
You can do that with the following method:

```C#
using (var image = new MagickImage("without.tif"))
{
    using (var alpha = new MagickImage("with.tif"))
    {
        image.Composite(alpha, CompositeOperator.CopyAlpha);

        image.Write("out.tif");
    }
}
```

**Reply by adobris** [reply]
Thank you!
